### PR TITLE
Use virtualenv for installing Renku

### PR DESCRIPTION
**Pull request by pameladelgado**
- use virtualenv instead of pipx for installing Renku (see issue with setuptools https://renku.discourse.group/t/error-during-environment-creation/407/6) 
- update renku version

- test projects created in https://renkulab.io/projects/pamela.delgado/test-streamlit-template-fixed and 
https://renkulab.io/projects/pamela.delgado/test-aiida-template-fixed, images build properly and the newer version of renku is installed

### CODE REVIEW

File: aiida/Dockerfile
Comment (ableuler) [inline aiida/Dockerfile:42]: Simply using pip install here will install renku python together with all the other python packages which comes at a risk of colliding common dependencies. Avoiding this was the intention behind using pipx, which installed renku python into its own virtual environment. Replicating pipx without using pipx was the idea behind what I added https://github.com/SwissDataScienceCenter/renku-project-template/blob/4068f96df5512c082e75766289dad6972a577f3e/python-minimal/Dockerfile#L37. For this approach to work here you'd have to bump the base image to `>=0.10.1`.

**Review by ableuler — CHANGES_REQUESTED** [review]

File: aiida/Dockerfile
Comment (pameladelgado) [reply aiida/Dockerfile:42]: > comes at a risk of colliding common dependencies

For the moment the images built with the templates dont seem to have any colliding dependencies issue.

>For this approach to work here you'd have to bump the base image to >=0.10.1.

People that maintain and use these templates might not want/be able to work with python 3.9 yet and unfortuntately we dont have an image with a `3.8-0.10.1` tag available yet.

Comment (ableuler) [reply aiida/Dockerfile:42]: > For the moment the images built with these templates dont seem to have any colliding dependencies issue.

I think the collisions are more likely to happen with packages that users will add to their "per-project" `requirements.txt`. I'd therefore try to stick to installing renku in a separate virtualenv. 

If upgrading the base image is not an option, you could create the virtualenv in this Dockerfile along the lines of what I did here: https://github.com/SwissDataScienceCenter/renkulab-docker/blob/4eed73105459025f45d4b854247e5b721b8c183b/docker/py/Dockerfile#L62

The other option of course is for us to stop messing with her image and to leave this decisions to @LiliGasser :)

Comment (LiliGasser) [reply aiida/Dockerfile:42]: The base image can be upgraded to a Python 3.9 one. 

Can I leave that to the two of you?

Comment (pameladelgado) [reply aiida/Dockerfile:42]: @LiliGasser I have updated the base image, could you to double check everything else is in order? and approve if so?
- streamlit https://renkulab.io/projects/pamela.delgado/test-streamlit-new-base-image
- aiida https://renkulab.io/projects/pamela.delgado/test-aiida-new-base-image

File: aiida/Dockerfile
Comment (ableuler) [inline aiida/Dockerfile:33]: I'd exchange everything below this line with the block below https://github.com/SwissDataScienceCenter/renku-project-template/blob/4068f96df5512c082e75766289dad6972a577f3e/python-minimal/Dockerfile#L27.

For sure we need to activate the right virtualenv before un(installing) the renku cli here.

**Review by ableuler — CHANGES_REQUESTED** [review]

**Review by LiliGasser — APPROVED** [review]
It looks good to me.

I am not sure whether @ableuler  comment on the Dockerfile should also be applied to the streamlit one.

Thanks, @pameladelgado !

File: aiida/Dockerfile
Comment (pameladelgado) [reply aiida/Dockerfile:33]: I do not think we need this, since the templates are not planned to be used with a dev CLI version
># Install renku from pypi or from github if it's a dev version

And the argument with the pinned renku version can stay that way too.

Comment (ableuler) [reply aiida/Dockerfile:33]: Ok, but even if we always keep it to `RENKU_VERSION=0.16.0` and never install a dev version from github, we _need_ to activate the virtualenv in which the renku cli is installed before pip uninstalling and pip re-installing it. Otherwise we might end up with two renkus, one in `/opt/conda/bin/renku` (installed in this Dockerfile) and one in `/home/jovyan/.renku/bin/renku` which comes with the image.

Comment (pameladelgado) [reply aiida/Dockerfile:33]: yep, fair enough. I'll add the venv part shortly.

**Review by ableuler — APPROVED** [review]
Thanks for fixing those templates!

**Pull request merged**
